python/ray/worker.py

```diff
@@ -2574,18 +2571,31 @@ def wait(object_ids, num_returns=1, timeout=None, worker=global_worker):
         if len(object_ids) == 0:
             return [], []
 
-        object_id_strs = [
-            plasma.ObjectID(object_id.id()) for object_id in object_ids
-        ]
+        if len(object_ids) != len(set(object_ids)):
+            raise Exception("Wait requires a list of unique object IDs.")
+        if len(object_ids) == 0:
+            raise Exception("Invalid number of objects %d." % len(object_ids))
```

Comment:
Actually, this particular check should only be necessary for legacy Ray, and not for xray, right?